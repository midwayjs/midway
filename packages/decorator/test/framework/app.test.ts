import {
  App,
  getClassMetadata,
  CLASS_KEY_CONSTRUCTOR,
  APPLICATION_KEY,
} from '../../src';

class Test {
  constructor(@App() aaa: any) {
    // ignore
  }

  @App()
  hhh: any;
}

describe('/test/framework/config.test.ts', () => {
  it('config decorator should be ok', () => {
    let data = getClassMetadata(CLASS_KEY_CONSTRUCTOR, Test);
    expect(data).toStrictEqual({
      0: {
        key: 'aaa',
        type: APPLICATION_KEY,
      },
    });

    data = getClassMetadata(APPLICATION_KEY, Test);
    expect(data).toStrictEqual([{ key: APPLICATION_KEY, propertyName: 'hhh' }]);
  });
});
