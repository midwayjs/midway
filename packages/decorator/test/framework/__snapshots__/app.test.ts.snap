// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/test/framework/app.test.ts app decorator should be ok 1`] = `
Object {
  "hhh": Object {
    "impl": true,
    "key": "__midway_framework_app__",
    "metadata": Object {
      "type": undefined,
    },
    "propertyName": "hhh",
  },
}
`;
